Add a generic description to the alg (#5998)
qgis/QGIS-Documentation: docs/user_manual/processing_algs/qgis/vectoranalysis.rst

Nearest neighbour analysis
--------------------------

> Performs nearest neighbor analysis for a point layer. The output allows to
> understand how your data are distributed (clustered, randomly or distributed).

Reviewer: ```suggestion
Performs nearest neighbor analysis for a point layer. The output tells you
how your data are distributed (clustered, randomly or distributed).
```

* Expected mean distance
* Nearest neighbour index
* Number of points
* Z-Score: Comparing the Z-Score with the normal distribution tells
  you how your data are distributed.

>   A low Z-Score means that the data are unlikely distributed randomly
>   while a high Z-Score means that your data are very likely
>   distributed in a dispersed way.

Reviewer: ```suggestion
  A low Z-Score means that the data are unlikely to be the result of a
  spatially random process, while a high Z-Score means that your data
  are likely to be a result of spatially random process.
```

Author: **are** (un)likely **to be**?

Reviewer: > **are** (un)likely **to be**?

?
